Add cancel print

ismailsunni opened the pull request:
Adding function to cancel a print

ger-benjamin commented on demo/demo.js:
```diff
@@ -20,6 +26,22 @@ const map = new Map({
   }),
 });
 
+var report = null;
```
```suggestion
let report = null;
```

ger-benjamin commented on demo/demo.js:
```diff
@@ -56,20 +78,24 @@ document.querySelector('#print').addEventListener('click', async () => {
+      },
+    )
+    .then(() => {
+      report = null;
```
In case of error, it will not pass here I think. Maybe it's better with a `finally`. (to test, it's easy with bad parameters, like removing the mandatory `datasource` at new line 69)

ger-benjamin reviewed (approved):
2 comments, but looks good, thanks :-)

ismailsunni commented:
Thanks @ger-benjamin I have addressed your review :)

gberaudo commented on demo/demo.js:
```diff
@@ -20,6 +26,22 @@ const map = new Map({
+let report = null;
+
+document.querySelector('#cancel').addEventListener('click', async () => {
+  console.log('Cancel print');
+  if (report) {
+    const cancelResult = await cancelPrint(MFP_URL, report.ref);
+    if (cancelResult.status === 200) {
+      console.log('Print is canceled');
+    } else {
+      console.log('Failed to cancel print');
+    }
+  } else {
+    console.log('No print in progress');
+  }
```
Instead of using console.log you should change content of the #result textarea.

gberaudo commented on src/utils.ts:
```diff
@@ -107,6 +107,10 @@ export async function getStatus(mfpBaseUrl: string, ref: string): Promise<MFPSta
 }
 
+export async function cancelPrint(mfpBaseUrl: string, ref: string): Promise<MFPCancelResponse> {
+  return await (await fetch(`${mfpBaseUrl}/cancel/${ref}`, {method: 'DELETE'})).json();
```
What about using several line?
That's a lot of `await`s here.

ismailsunni commented:
Thanks @gberaudo I have addressed your review :)
I am logging off now, feel free to merge it if you want to release a new version. I will be back on Monday unfortunately.
